### DISCUSSION TITLE
TikZ support: Getting coordinates from pdf viewer

### DISCUSSION THREAD

**Discussion by egidioln** [texstudio-org/texstudio#2307] [Q&A] [answered]
Hello! 
Is there any way to get the values of x, y coordinates for TikZ (maybe with `overlay` and `shift=(current page.center)` options)  from a click on the pdf viewer (as the "Go to Source" command does for code)? I think this would be very useful for drawing or placing nodes on slides with TikZ.

If not, I'd be willing to help implement something like this :)

**Comment by muzimuzhi** [comment, marked as answer]
I'm afraid that's impossible. In the output PDF, the tikz coordinate info is lost.

To ease the node positioning task, path operation `grid` (like `\draw (0,0) grid (3,2);`, see pgfmanual sec. [14.8](https://tikz.dev/tikz-paths#sec-14.8)) and relative node positioning (see pgfmanual sec. [17.5.2](https://tikz.dev/tikz-shapes#sec-17.5.2) and [17.5.3](https://tikz.dev/tikz-shapes#sec-17.5.3)) may help.

**Reply by egidioln** [reply]
Thanks for your reply @muzimuzhi ! Indeed, I'm using `grid` for that now but some interactive method would be really helpful... 

Would you know if there is at least a way to recover x, y position of a mouse click relative to the page corner (in cm, for instance)? This would help to draw in standalone documents (with `\documentclass[tikz]{standalone}`) and then move the final figure to the original document.

**Reply by muzimuzhi** [reply]
Sorry I don't know. Searching for "pdf reader, show mouse coordinate" shows [Adobe Acrobat](https://community.adobe.com/t5/acrobat-reader-discussions/cursor-coordinates-option-is-not-appear-in-acrobat-dc/td-p/10427433) and some other readers (see https://stackoverflow.com/q/8971243) have this utility. Then the task becomes, how to config texstudio to work with outer pdf reader.

Also there's some indirect ways to draw grid with coordinate label ticks, see https://github.com/pgf-tikz/pgf/issues/760.

**Reply by muzimuzhi** [reply]
In case it becomes an X-Y problem, I'd suggest to post on tex.stackexchange.com with an example of your real use case.

**Reply by egidioln** [reply]
Thank you for your suggestions :) 
I'll try something like this
